redis/src/commands/mod.rs

```diff
@@ -1420,6 +1420,53 @@ implement_commands! {
         cmd("XADD").arg(key).arg(id).arg(map)
     }
 
+
+    /// Add a stream message with options.
+    ///
+    /// ```text
+    /// XADD key [NOMKSTREAM] [<MAXLEN|MINID> [~|=] threshold [LIMIT count]] <* | ID> field value [field value] ...
+    /// ```
+    #[cfg(feature = "streams")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "streams")))]
+    fn xadd_options<
+        K: ToRedisArgs, ID: ToRedisArgs, F: ToRedisArgs, V: ToRedisArgs
+    >(
+        key: K,
+        id: ID,
+        items: &'a [(F, V)],
+        options: &'a streams::StreamAddOptions
```

Comment:
@nihohit I have a question about this here.   currently there is an inconsistency in how options params are passed in. Sometimes they are passed by ref, and other times they are passed as owned.  What is the preference for the new methods I am adding in here (and what the API should morph to going forward)?

(examples of fns using owned objects for options are geo_radius* or xclaim_options)

Reply:
Let's use references here. I don't know why some options are owned 🤷